code/modules/projectiles/guns/energy/disintegrator.dm

```diff
@@ -98,32 +98,11 @@
 	projectile_type = "/obj/item/projectile/beam/scorchray/atomizationray"
 	origin_tech = Tc_COMBAT + "=5;" + Tc_MATERIALS + "=3" + Tc_POWERSTORAGE + "=4"
 	fire_delay = 0.6 SECONDS // Barely noticeable, mostly here to allow the firing noise .ogg to finish ~0.55 seconds
-
-	var/charge_tick = 0
-	var/charge_wait = 4 // This one charges itself like the Captain's laser, at the cost of fun alternate modes
+	recharge_time = 4 // This one charges itself like the Captain's laser, at the cost of fun alternate modes
+	recharge_mult = 2
```

Comment:
Note this appears to be a functional change (which is that the gun doesn't drain energy while recharging). This is fine, but should be CL noted (all affected guns).

Reply:
was not intended as such

Reply:
It looks like all guns you edited here (just the next few files) all are now recharging but don't drain energy as they do so. They used to have this hard-coded, though frankly there is a more OOP way to implement this, which is that guns types designed to be modules (and thus auto-recharge) should just have a shared variable that handles this in process. Alternatively, you could have the robot's process() check for guns inside and give them charge if they are not full, which would flexibly work for all gun modules.

Reply:
still not really seeing it because i intended to keep functionality 1:1, would help if you could review comment more specifically which place has been changed. also whose energy? borgs?